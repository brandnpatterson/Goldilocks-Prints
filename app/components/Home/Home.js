import React from 'react'
import './Home.sass'

const Home = () => {
  return (
    <div className='home'>
      <div className='panel'>
        <h1>Home</h1>
      </div>
    </div>
  )
}

export default Home
