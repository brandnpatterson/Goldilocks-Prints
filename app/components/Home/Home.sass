@import ../../index.sass

.home
  .landing
    font-size: 25px
    +mediaQuery($small)
      font-size: 15px
    text-align: center
  .panel
    background: $white
    height: 300px
    padding: 50px 0
    text-align: center
    width: 100vw
