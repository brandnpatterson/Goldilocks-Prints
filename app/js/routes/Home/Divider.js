import React, { Component } from 'react'
import { MdAirplanemodeActive } from 'react-icons/lib/md/'

import '../../styles/Divider.sass'

const { string } = React.PropTypes

export default class Divider extends Component {
  render () {
    return (
      <div className='divider'>
        <MdAirplanemodeActive className='icon' />
        <h2 className='divider-text'>{this.props.text}</h2>
      </div>
    )
  }
}

Divider.propTypes = {
  text: string
}
