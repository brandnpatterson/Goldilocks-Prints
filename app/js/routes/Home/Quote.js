import React, { Component } from 'react'

const opacityArr = [ 0, 0.6, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.6, 0.3, 0, 0, 0 ]

export default class Quote extends Component {
  constructor () {
    super()
    this.state = {
      opacity: opacityArr[0],
      quote: '"Imagine the possibilities..."'
    }
  }

  componentDidMount () {
    let opacityPosition = 0
    setInterval(() => {
      if (opacityArr.length - 1 > opacityPosition) {
        this.setState({
          opacity: opacityArr[opacityPosition]
        })
        opacityPosition++
      } else {
        this.setState({
          opacity: opacityArr[opacityPosition]
        })
        opacityPosition = 0
      }
    }, 500)
  }

  render () {
    const quoteStyle = {
      color: '#fff',
      fontSize: '1.5em',
      opacity: this.state.opacity,
      position: 'absolute',
      left: 20,
      top: 20
    }
    return (
      <div>
        <h2 style={quoteStyle}>{this.state.quote}</h2>
      </div>
    )
  }
}
