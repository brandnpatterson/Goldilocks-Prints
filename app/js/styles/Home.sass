.home
  padding: 0

  img
    width: 100vw
