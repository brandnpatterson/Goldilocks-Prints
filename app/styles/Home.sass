body
  background: url('../../public/images/landing.jpeg') center no-repeat fixed
  background-size: cover
  margin: 0
  width: 100vw

.home
  padding: 0

  img
    width: 100vw
