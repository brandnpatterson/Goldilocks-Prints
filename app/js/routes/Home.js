import React, { Component } from 'react'

import Section from './Home/Section'

import '../styles/Home.sass'

export default class Home extends Component {
  render () {
    return (
      <div className='home'>
        <Section
          className='section'
          backgroundImage='url(../../images/landing.jpeg)'
        />
      </div>
    )
  }
}
