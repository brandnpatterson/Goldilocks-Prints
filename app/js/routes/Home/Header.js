import React, { Component } from 'react'

import Quote from './Quote'
const imageArr = [ 'public/images/landing1.jpeg', 'public/images/landing2.jpeg', 'public/images/landing1.jpeg' ]

export default class Header extends Component {
  constructor () {
    super()
    this.state = {
      image: imageArr[0]
    }
  }

  componentDidMount () {
    let imagePosition = 0
    setInterval(() => {
      if (imageArr.length - 1 > imagePosition) {
        this.setState({
          image: imageArr[imagePosition]
        })
        imagePosition++
      } else {
        this.setState({
          image: imageArr[imagePosition]
        })
        imagePosition = 0
      }
    }, 4000)
  }

  render () {
    const headerStyle = {
      width: '100vw'
    }
    return (
      <div>
        <img style={headerStyle} src={this.state.image} alt='header images slideshow' />
        <Quote />
      </div>
    )
  }
}
