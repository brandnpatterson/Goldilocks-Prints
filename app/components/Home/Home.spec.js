import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Home from './Home'

test('It should render the Home component as expected', () => {
  const component = shallow(<Home />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
