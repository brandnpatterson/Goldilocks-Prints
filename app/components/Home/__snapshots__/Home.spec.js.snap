// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the Home component as expected 1`] = `
<div
  className="home"
>
  <div
    className="landing"
  >
    <h1>
      Goldilox Prints.
    </h1>
  </div>
  <div
    className="panel"
  >
    <h1>
      Home
    </h1>
  </div>
</div>
`;
